src/main/java/com/minecolonies/coremod/network/messages/ReplaceBlockMessage.java

```diff
@@ -0,0 +1,202 @@
+                        }
+                        else if(newBlockState.getBlock() instanceof BlockLog && blockState.getBlock() instanceof BlockLog)
+                        {
+                            final IBlockState transformation = newBlockState.withProperty(BlockLog.LOG_AXIS, blockState.getValue(BlockLog.LOG_AXIS));
+                            world.setBlockState(here, transformation);
+                        }
+                        else if(newBlockState.getBlock() instanceof BlockRotatedPillar && blockState.getBlock() instanceof BlockRotatedPillar)
+                        {
+                            final IBlockState transformation = newBlockState.withProperty(BlockRotatedPillar.AXIS, blockState.getValue(BlockRotatedPillar.AXIS));
+                            world.setBlockState(here, transformation);
+                        }
+                        else if(newBlockState.getBlock() instanceof BlockTrapDoor && blockState.getBlock() instanceof BlockTrapDoor)
+                        {
+                            IBlockState transformation = newBlockState.withProperty(BlockTrapDoor.HALF, blockState.getValue(BlockTrapDoor.HALF));
+                            transformation = transformation.withProperty(BlockTrapDoor.FACING, blockState.getValue(BlockTrapDoor.FACING));
+                            transformation = transformation.withProperty(BlockTrapDoor.OPEN, blockState.getValue(BlockTrapDoor.OPEN));
+                            world.setBlockState(here, transformation);
+                        }
+                        else if(newBlockState.getBlock() instanceof BlockDoor && blockState.getBlock() instanceof BlockDoor)
+                        {
+                            final IBlockState transformation = newBlockState.withProperty(BlockDoor.FACING, blockState.getValue(BlockDoor.FACING));
+                            world.setBlockState(here, transformation);
+                        }
+                        else if (stackToPlace.getItem() == Items.LAVA_BUCKET)
+                        {
+                            world.setBlockState(here, Blocks.LAVA.getDefaultState());
+                        }
+                        else if (stackToPlace.getItem() == Items.WATER_BUCKET)
+                        {
+                            world.setBlockState(here, Blocks.WATER.getDefaultState());
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Is this the correct block to remove it or replace it.
+     * @param worldStack the world stack to check.
+     * @param worldState the world state to check.
+     * @param compareStack the comparison stack.
+     * @return true if so.
+     */
+    public static boolean correctBlockToRemoveOrReplace(final ItemStack worldStack, final IBlockState worldState, final ItemStack compareStack)
+    {
+        return worldStack != null && (worldStack.isItemEqual(compareStack)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (7) used in the expression (maximum allowed 6). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)